### PR TITLE
1622

### LINKED ISSUE #1622: Error with required fields when creating a listing
When creating a listing with all fields blank, only tags and description field are marked in red. Error message informs that only Tags field is required:

_Data could not be saved:_ 
_Some fields are missing or incorrect._
_\- Tags_

check required

Then, if user fills tags field, error message informs other required fields, that now are marked in red:

_\- Title_
_\- Price_
_\- Description_
_\- Processing Time_

check required2

And if those fields are filled, then, error message informs the last required field, that now is NOT marked in red:

_\- images_

Other problem:
If left blank, "Ships To" field display on error message as "undefined".

**Expected Behavior:**
All required fields should be marked in red (including Images/Photos field) and error message should include all missing required fields at once, and on the correct order of appearance. "images" item on error message should be renamed to "Photos":

_Data could not be saved:_ 
_Some fields are missing or incorrect._

_\- Title_
_\- Price_
_\- Description_
_\- Processing Time_
_\- Ships To_
_\- Photos_
_\- Tags_

Operating System (OS and version): Win7 64 bits
OpenBazaar version: 1.1.5
Reproducible: Always

### PR DESCRIPTION
**Pull request by rmisio**
closes #1622

### CODE REVIEW

File: js/views/itemEditVw.js [hunk 1]
```diff
@@ -194,7 +192,13 @@ module.exports = baseVw.extend({
         tags: keywordTags,
         preserveCase: true,
         saveOnBlur: true,
-        placeholder: window.polyglot.t('KeywordsPlaceholder')
+        placeholder: window.polyglot.t('KeywordsPlaceholder'),
+        onTagAdd: () => {
+          this.$('#inputKeyword').removeClass('invalid');
+        },
+        onTagRemove: () => {
+          this.$('#inputKeyword').addClass('invalid');
```
Comment (jjeffryes) [inline js/views/itemEditVw.js]: This marks the field as invalid if you have 3 tags, and delete one, leaving 2 tags. 

You could put in a count that checks to see if there are zero tags when the remove event fires.

File: js/views/itemEditVw.js [hunk 2]
```diff
@@ -603,13 +617,21 @@ module.exports = baseVw.extend({
...
+
+    $(submitForm).add(this.$('#imageForm')).find(':invalid, .invalid').each(function() {
+      var inputName,
+          $label;
+
+      inputName = (($label = self.$("label[for='"+$(this).attr('id')+"']")).length && $label.text()) ||
+        $(this).attr('data-label') || $(this).attr('id');
+      inputName = inputName.trim();
+
+      invalidInputList.push(inputName.trim())
```
Comment (jjeffryes) [inline js/views/itemEditVw.js:630]: Does inputName need .trim() twice?

File: js/views/itemEditVw.js (hunk 1 above)
Comment (rmisio) [inline js/views/itemEditVw.js]: good catch. will fix shortly.

File: js/views/itemEditVw.js (hunk 2 above)
Comment (rmisio) [inline js/views/itemEditVw.js:630]: absolutely not. will fix.

**Comment by rmisio** [comment]
Issues from peer review have been fixed.

**Pull request merged**